server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala

```diff
@@ -76,10 +117,16 @@ class NettyServerHandler[F[_]](route: Route[F], unsafeRunAsync: (() => F[Unit])
       request match {
         case full: FullHttpRequest =>
           val req = full.retain()
-          unsafeRunAsync { () =>
+          val cancellationSwitch: () => Future[Unit] = unsafeRunAsync { () =>
             runRoute(req)
-              .ensure(me.eval(req.release()))
+              .ensure {
+                me.eval {
+                  pendingResponses.dequeue()
```

Comment:
doesn't this make the assumption that the requests will end in the same order, in which they started?

Reply:
They are requests from the same channel, so maybe that's OK. Anyway, in http4s implementation there additional elements like `eventLoopContext` and `pendingResponses`. I think they are crucial to make sure this ordering works as expected, so I'll study them further.